pkg/operator/configobservation/apiserver/observer_apiserver_test.go

```diff
@@ -0,0 +1,415 @@
+				t.Errorf("did not expect observed config to be updated : %s", result)
+			}
+			if !equality.Semantic.DeepEqual(tc.expectedSynced, synced) {
+				t.Errorf("expected resources not synced: %s", diff.ObjectReflectDiff(tc.expectedSynced, synced))
+			}
+		})
+	}
+}
+
+func TestObserveDefaultServingCertificate(t *testing.T) {
+
+	existingConfig := map[string]interface{}{
+		"servingInfo": map[string]interface{}{
+			"certFile": "/etc/kubernetes/static-pod-resources/secrets/existing/tls.key",
+		},
+	}
+
+	testCases := []struct {
+		name           string
+		config         *configv1.APIServer
+		expected       map[string]interface{}
+		expectedSynced map[string]string
+	}{
+		{
+			name:     "NoAPIServerConfig",
+			config:   nil,
+			expected: map[string]interface{}{},
+		},
+		{
+			name:     "NoUserServingCertRef",
+			config:   newAPIServerConfig(),
+			expected: map[string]interface{}{},
+		},
+		{
+			name:   "HappyPath",
+			config: newAPIServerConfig(withDefaultSecret("happy")),
+			expected: map[string]interface{}{
+				"servingInfo": map[string]interface{}{
+					"certFile": "/etc/kubernetes/static-pod-resources/secrets/user-serving-cert/tls.crt",
+					"keyFile":  "/etc/kubernetes/static-pod-resources/secrets/user-serving-cert/tls.key",
+				},
+			},
+			expectedSynced: map[string]string{
+				"configmap/user-serving-cert.openshift-kube-apiserver-operator": "configmap/happy.openshift-config",
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			indexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{})
+			if tc.config != nil {
+				if err := indexer.Add(tc.config); err != nil {
+					t.Fatal(err)
+				}
+			}
+			synced := map[string]string{}
+			listers := configobservation.Listers{
+				APIServerLister: configlistersv1.NewAPIServerLister(indexer),
+				ResourceSync:    &mockResourceSyncer{t: t, synced: synced},
+			}
+			result, errs := ObserveUserServingCertificate(listers, events.NewInMemoryRecorder(t.Name()), existingConfig)
+			if len(errs) > 0 {
+				t.Errorf("Expected 0 errors, got %v.", len(errs))
+			}
+			if !equality.Semantic.DeepEqual(tc.expected, result) {
+				t.Errorf("result does not match expected config: %s", diff.ObjectDiff(tc.expected, result))
+			}
+			if !equality.Semantic.DeepEqual(tc.expectedSynced, synced) {
+				t.Errorf("expected resources not synced: %s", diff.ObjectReflectDiff(tc.expectedSynced, synced))
+			}
+		})
+	}
+}
+
+func TestObserveNamedCertificates(t *testing.T) {
+
+	existingConfig := map[string]interface{}{
+		"servingInfo": map[string]interface{}{
+			"namedCertificates": []interface{}{
+				map[string]interface{}{
+					"certFile": "/etc/kubernetes/static-pod-resources/secrets/existing/tls.crt",
+					"keyFile":  "/etc/kubernetes/static-pod-resources/secrets/existing/tls.key",
+					"names":    []interface{}{"existing"},
+				},
+			},
+		},
+	}
+
+	testCases := []struct {
+		name           string
+		config         *configv1.APIServer
```

Comment:
same initial static comment here.

Reply:
`existingConfig` defined above is used as the initial state in all testcases.

Reply:
> `existingConfig` defined above is used as the initial state in all testcases.

there are different existing states.  I don't mind the definition above, just that you need to be able to specify it per test and you can reference the your `existingConfig` variable for most of them.  Clearly you should have an empty starting point too at the very least.